### PR TITLE
Ship tcl and tk libs and get em signed up for macOS

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

Ship tcl and tk on mac and get em signed up.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/CMakeLists.txt
Comment (jmarrec) [inline src/EnergyPlus/CMakeLists.txt:973]: Call this script after pip installing energyplus-launch

File: cmake/PythonFixUpTclTk.py
Comment (jmarrec) [inline cmake/PythonFixUpTclTk.py]: The script takes the path to python_lib in Products/ dir and will:
* Locate the linked libs for tcl and tk
* copy them next to _tkinter so
* install_name_tool to change the link to these copied libs for _tkinter
* (change the tcl and tk lib ids, I got confused by them)

Maybe it shouldn't have been a python script and a cmake one instead, but we already have PythonCopyStandardLib so whatever. I went with easiest.

File: cmake/install_codesign_script.cmake
Comment (jmarrec) [inline cmake/install_codesign_script.cmake:116]: And when we glob the python .sos, we also glob the tcl and tk dylibs so they get signed as well.

```
/Users/julien/Software/Others/EnergyPlus-build-release/_CPack_Packages/Darwin/TGZ/EnergyPlus-24.2.0-cdf84ccb68-Darwin-macOS12.1-arm64/energyplus auxiliary eplaunch
```

image


the end.

File: cmake/PythonCopyStandardLib.py
Comment (Myoldmopar) [inline cmake/PythonCopyStandardLib.py:125]: Yes

File: cmake/PythonFixUpOnMac.cmake
Comment (Myoldmopar) [inline cmake/PythonFixUpOnMac.cmake:90]: Yes

File: cmake/PythonFixUpTclTk.py
Comment (Myoldmopar) [inline cmake/PythonFixUpTclTk.py]: Tempted to change this string.

**Review by Myoldmopar — APPROVED** [review]
Fantastic!

**Comment by Myoldmopar** [comment]
OK, so Mac failed on arm64 because the otool output didn't have the compatability stuff, it just says 'architecture arm64'.  I added a second RE matcher and I think that will be fine.  I added license text to fix up the integrity check.  I'll wait until #10737 since it will likely merge in a few minutes and pull that in, then re test it here and another test package.

**Comment by Myoldmopar** [comment]
Alright, let's try this again!  I will know whether this is fixed in like 37 minutes, so I'll wait until then to kick off a release package so that it doesn't fail twice.

**Comment by Myoldmopar** [comment]
Dang, now a different problem.  I guess that's still progress.... testing locally to debug.

**Comment by Myoldmopar** [comment]
There's a regex syntax warning showing up:
```
/Users/runner/work/EnergyPlus/EnergyPlus/branch/cmake/PythonFixUpTclTk.py:75: SyntaxWarning: invalid escape sequence '\('
  LINKED_RE_ARM64 = re.compile(f"(?P<libname>.*) \(architecture arm64\)")
```


After this part, it gets a permission error.  I don't _think_ it's related...but I'm not sure.

```
  File "/Users/runner/work/EnergyPlus/EnergyPlus/branch/cmake/PythonFixUpTclTk.py", line 116, in <module>
    shutil.copy(tcl_tk_so, new_tcl_tk_so)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/shutil.py", line 435, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/shutil.py", line 262, in copyfile
    with open(dst, 'wb') as fdst:
         ^^^^^^^^^^^^^^^
PermissionError: [Errno 13] Permission denied: '/Users/runner/work/EnergyPlus/EnergyPlus/branch/build/Products/python_lib/lib-dynload/libtcl8.6.dylib'
```

I fixed that regex locally and it doesn't change anything.  But locally, I'm actually getting a failure that it cannot find the _tkinter so file at all.  I've brew installed python-tk but it's still not there.

I'm a bit confused at the moment but will keep trying things in a bit.

**Comment by Myoldmopar** [comment]
When I brew install python, and even python-tk, I don't seem to get the _tkinter libraries that our build rules expect to find.  I know there are lots of issues about this around the web, and I haven't nailed down the exact solution I need.  So locally I am getting a different problem, and can't reproduce this issue on GHA right now. I'll keep trying...

**Comment by jmarrec** [comment]
> There's a regex syntax warning showing up:
> ```
> /Users/runner/work/EnergyPlus/EnergyPlus/branch/cmake/PythonFixUpTclTk.py:75: SyntaxWarning: invalid escape sequence '\('
>   LINKED_RE_ARM64 = re.compile(f"(?P<libname>.*) \(architecture arm64\)")
> ```
> 
> 
> After this part, it gets a permission error.  I don't _think_ it's related...but I'm not sure.
> 
> ```
>   File "/Users/runner/work/EnergyPlus/EnergyPlus/branch/cmake/PythonFixUpTclTk.py", line 116, in <module>
>     shutil.copy(tcl_tk_so, new_tcl_tk_so)
>   File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/shutil.py", line 435, in copy
>     copyfile(src, dst, follow_symlinks=follow_symlinks)
>   File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/shutil.py", line 262, in copyfile
>     with open(dst, 'wb') as fdst:
>          ^^^^^^^^^^^^^^^
> PermissionError: [Errno 13] Permission denied: '/Users/runner/work/EnergyPlus/EnergyPlus/branch/build/Products/python_lib/lib-dynload/libtcl8.6.dylib'
> ```
> 
> I fixed that regex locally and it doesn't change anything.  But locally, I'm actually getting a failure that it cannot find the _tkinter so file at all.  I've brew installed python-tk but it's still not there.
> 
> I'm a bit confused at the moment but will keep trying things in a bit.

That permission error looks like shutil.copy trying to copy on an file that already exists.

**Comment by Myoldmopar** [comment]
Alright, let's see!  If CI comes back happy on Mac arm64 in the next 40 minutes or so, I'll kick off a release build, and we'll go from there.

**Comment by Myoldmopar** [comment]
A couple paper cuts I'd like to solve but probably won't until later:
- One related to this work.  If I run energyplus.exe auxiliary eplaunc, it doesn't complain about the misspelled auxiliary tool, it just says I need to enter a command
- One unrelated, when I install on Windows, if I *uncheck* the start menu option in the installer, it still asks me for a start menu folder name.

Neither are big deals, just little paper cuts.

**Comment by Myoldmopar** [comment]
Pesky r string instead of f string.  Now I'm hopeful.

**Comment by Myoldmopar** [comment]
It worked!! @jmarrec it worked!!  Merging this in.  I'll take a little time shortly to update the README and other stuff and then tag RC3 for final testing.

File: cmake/PythonFixUpTclTk.py
Comment (jmarrec) [inline cmake/PythonFixUpTclTk.py:78]: I'm confused about why this would be needed.

The verify_signature does not seem to have this issue

https://github.com/NREL/EnergyPlus/blob/30b074af220892f2009d830f564725f365e258c2/scripts/dev/verify_signature.py#L150-L153

Can I reproduce the problem you were seeing?

**Pull request merged**
